pkg/controller/util/typeconfig.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import "github.com/kubernetes-sigs/federation-v2/pkg/apis/core/typeconfig"
+
+func IsNamespaceKind(t typeconfig.Interface) bool {
```

Comment:
Should this method be added to the interface?

Reply:
I originally thought and wanted to do this, but I don't want to rely on the `util.NamespaceKind` constant within the `v1alpha1` package nor hard code the string `Namespace` again. But I can add it there instead if you'd prefer, and either move the common util constant there (many things depend on it) or just hard-code it again.

Reply:
+1 to adding it to the interface (and an implementation to `FederatedTypeConfig`).  There is already an implementation of checking that the target is a namespace that could be refactored into the implementation:

https://github.com/kubernetes-sigs/federation-v2/blob/master/pkg/apis/core/v1alpha1/federatedtypeconfig_types.go#L215